src/main/java/org/mifos/connector/mockpaymentschema/service/AirtelMockService.java

```diff
@@ -0,0 +1,158 @@
+package org.mifos.connector.mockpaymentschema.service;
+
+import java.util.UUID;
+import org.mifos.connector.common.airtel.dto.AirtelEnquiryResponseDTO;
+import org.mifos.connector.common.airtel.dto.AirtelEnquiryResponseDataDTO;
+import org.mifos.connector.common.airtel.dto.AirtelEnquiryResponseDataTransactionDTO;
+import org.mifos.connector.common.airtel.dto.AirtelPaymentRequestDTO;
+import org.mifos.connector.common.airtel.dto.AirtelPaymentResponseDTO;
+import org.mifos.connector.common.airtel.dto.AirtelPaymentResponseDataDTO;
+import org.mifos.connector.common.airtel.dto.AirtelPaymentResponseDataTransactionDTO;
+import org.mifos.connector.common.airtel.dto.AirtelResponseStatusDTO;
+import org.mifos.connector.mockpaymentschema.schema.TransferStatus;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+
+@Service
+public class AirtelMockService {
+
+    @Autowired
+    private org.mifos.connector.mockpaymentschema.service.SendCallbackService sendCallbackService;
+
+    @Value("${airtel.msisdnSuccessful}")
+    private String msisdnSuccessful;
+
+    @Value("${airtel.msisdnFailed}")
+    private String msisdnFailed;
+
+    @Value("${airtel.transactionIdSuccessful}")
+    private String transactionIdSuccessful;
+
+    @Value("${airtel.transactionIdFailed}")
+    private String transactionIdFailed;
+
+    @Value("${airtel.successResponseCode}")
+    private String successResponseCode;
+
+    @Value("${airtel.failedResponseCode}")
+    private String failedResponseCode;
+
+    @Value("${airtel.pendingResponsCode}")
+    private String pendingResponseCode;
+
+    public ResponseEntity<AirtelEnquiryResponseDTO> getTransactionStatus(String transactionId) {
+        String airtelMoneyId = UUID.randomUUID().toString().replace("-", "");
+        String message;
+        String status;
+        String code;
+        String responseCode;
+        boolean success;
+        HttpStatus httpStatus;
+
+        if (transactionId.equals(transactionIdSuccessful)) {
+            message = TransferStatus.SUCCESSFUL.name();;
+            status = "TS";
+            code = "200";
+            responseCode = successResponseCode;
+            success = true;
+            httpStatus = HttpStatus.OK;
+        }
+
+        else if (transactionId.equals(transactionIdFailed)) {
+            message = TransferStatus.FAILED.name();
+            status = "TF";
+            code = "400";
+            responseCode = failedResponseCode;
+            success = false;
+            httpStatus = HttpStatus.BAD_REQUEST;
+        }
+
+        else {
+            message = TransferStatus.IN_PROGRESS.name();
+            status = "TP";
```

Comment:
Instead of hardcoding for status and codes you can use enums

Reply:
Done.